## Check the last slide

In [ ]:
import json
import requests
from bs4 import BeautifulSoup
import time
import datetime as dt
import threading

def zodiac(day, month):
    
    if month == 1 and day >= 20 and day <=31 or month == 2 and day >= 1 and day <= 18:
        return f"Zodiac sign: Aquarius"
    elif  month == 2 and day >= 19 and day <= 29 or month == 3 and day >= 1 and day <= 20:
        return f"Zodiac sign: Pisces"
    elif  month == 3 and day >= 21 and day <= 31 or month == 4 and day >= 1 and day <= 19:
        return f"Zodiac sign: Aries"
    elif  month == 4 and day >= 20 and day <= 30 or month == 5 and day >= 1 and day <= 20:
        return f"Zodiac sign: Taurus"
    elif  month == 5 and day >= 21 and day <= 31 or month == 6 and day >= 1 and day <= 20:
        return f"Zodiac sign: Gemini"
    elif  month == 6 and day >= 21 and day <= 30 or month == 7 and day >= 1 and day <= 22:
        return f"Zodiac sign: Cancer"
    elif  month == 7 and day >= 23 and day <= 31 or month == 8 and day >= 1 and day <= 22:
        return f"Zodiac sign: Leo"
    elif  month == 8 and day >= 23 and day <= 31 or month == 9 and day >= 1 and day <= 22:
        return f"Zodiac sign: Virgo"
    elif  month == 9 and day >= 23 and day <= 30 or month == 10 and day >= 1 and day <= 22:
        return f"Zodiac sign: Libra"
    elif  month == 10 and day >= 23 and day <= 31 or month == 11 and day >= 1 and day <= 21:
        return f"Zodiac sign: Scorpio"
    elif  month == 11 and day >= 22 and day <= 30 or month == 12 and day >= 1 and day <= 21:
        return f"Zodiac sign: Sagittarius"
    elif  month == 12 and day >= 22 and day <= 31 or month == 1 and day >= 1 and day <= 19:
        return f"Zodiac sign: Capricorn"

day = int(input("Enter day: "))
month = int(input("Enter month: "))
print(zodiac(day, month))

In [ ]:
import json
import requests
from bs4 import BeautifulSoup
import time
import datetime as dt
import threading
import pandas as pd

def percentage(day, month):
    print(day, month) 
    table = pd.read_csv("data.csv", sep = ",")
    print(table.head(5))
    print("----------------------------------------------------------------------------")
    total = table.count()
    maanden = table[table["date_of_month"] == day]
    lijst = maanden[maanden["month"] == month]
    sum_date_births = lijst["births"].sum()
    print(sum_date_births)
    
    sum_births = table["births"].sum()
    print(sum_births)
    percentage = (sum_date_births/ sum_births) * 100
    print(percentage)
    rounded = round(percentage,2)
    print(rounded)
    return f"Percentage of people born on same day: {rounded} %"
    
    
day = int(input("Enter day: "))
month = int(input("Enter month: "))
print(percentage(day, month))

In [ ]:
import json
import requests
from bs4 import BeautifulSoup
import time
import datetime as dt
import threading
import pandas as pd

def historical(day, month):
 
    url = f"https://api.api-ninjas.com/v1/historicalevents?month={month}&day={day}"
    headers = {"X-Api-Key" : "bncXEh6rAIsf7zRiKHR3BFJiilzJskTfH5u4Azus"}
    response = requests.get(url , headers=headers)

    if response.status_code == 200:
        data = response.json()
        
        output = "Historical events:"
        for x in data:
            year = x["year"]
            event = x["event"]
            outcome = f"\n{year} - {event}"
            output += outcome
        return output
            

day = int(input("Enter day: "))
month = int(input("Enter month: "))
print(historical(day, month))

In [42]:
import json
import requests
from bs4 import BeautifulSoup
import time
import datetime as dt
import threading
import pandas as pd
import re


def one(day, month, year):
    day = int(day)
    month = int(month)
    year = int(year)
    print(day, month, year)
    
    date = dt.datetime(year, month, day).strftime("%d %B %Y")
    print(date)
    dated = date.split()
    month = dated[1].lower()
    
    url = f"https://www.onthisday.com/date/{year}/{month}/{day}"
    
    headers = {"User-Agent": "Anthony"}
    response = requests.get(url, headers=headers )
    #print(response.text)
    print(response.status_code)
    
    soup = BeautifulSoup(response.text, "html.parser")
    evenementen = soup.find_all("li", class_="event")
    
    #print(evenementen)
    n = 1
    for e in evenementen:
        text = e.get_text(" ")
        image = e.find("img")
        
        if image:
            image.get("src")   
            if image.get("alt") == "US":
                output = text[11:].replace("  "," ")
                print(output)
                
                print(len(output))
                break
                n += 1
 
day = input("Enter day: ")
month = input("Enter month: ")
year = input("Enter year: ")
print(one(day, month, year))

5 8 2000
05 August 2000
200
 It's Gonna Be Me - 'N Sync
27
None


In [13]:
import requests
from bs4 import BeautifulSoup

url = "https://www.onthisday.com/date/2000/august/5"
headers = {"User-Agent": "Mozilla/5.0"}

res = requests.get(url, headers=headers)
soup = BeautifulSoup(res.text, "html.parser")
#print(res.text)
events = soup.find_all("li", class_="event")

for e in events:
    text = e.get_text(" ")
    #print(text)

In [ ]:
import requests
from bs4 import BeautifulSoup
import datetime as dt
import time


def wikipedia(day, month):
    print(day, month)
    
    date = dt.datetime(2026, month, day)
    dated = date.strftime("%d %B %Y")
    print(dated)
    parts = dated.split()
    month = parts[1]
    print(f"MONTH inputed = {month}\nDAY inputed = {day}\n ")
    
    url = f"https://en.wikipedia.org/wiki/{month}_{day}"
    headers = {"User-Agent": "Anthony"}
    response = requests.get(url, headers=headers)
    
    print(response.status_code)
    soup = BeautifulSoup(response.content, "html.parser")
    births = soup.find_all("li")
    n = 1
    for ul in births:
        print(ul.get_text())
        
day = 23
month = 4
print(wikipedia(day,month))


In [ ]:
import requests
from bs4 import BeautifulSoup
import datetime as dt
import time
print(time.ctime())#transforming unix epoch to actual date
print()

Sun Apr 26 21:27:36 2026



In [6]:
import requests
from bs4 import BeautifulSoup

url = "https://en.wikipedia.org/wiki/April_23"
response = requests.get(url, headers={"User-Agent": "Mozilla/5.0"})
soup = BeautifulSoup(response.text, "html.parser")

# id is directly on the <h2>
births_header = soup.find("h2", id="Births")

if births_header is None:
    raise ValueError("Could not find Births section")

entries = []
for sibling in births_header.parent.find_next_siblings():
    if sibling.name and "mw-heading2" in sibling.get("class", []):
        break
    for li in sibling.find_all("li"):
        text = li.get_text(strip=True)
        parts = text.split("–", 1)
        if len(parts) == 2:
            year_str = parts[0].strip().split()[0].strip()
            try:
                year = int(year_str)
                if 1141 <= year <= 1720:
                    entries.append({"year": year, "text": text})
            except ValueError:
                pass

for e in entries:
    print(f"{e['year']}: {e['text']}")

1185: 1185–Afonso II of Portugal(died 1223)[20]
1408: 1408–John de Vere, 12th Earl of Oxford(died 1462)[21]
1420: 1420–George of Poděbrady, King of Bohemia (died 1471)
1464: 1464–Joan of France, Duchess of Berry(died 1505)
1464: 1464   –Robert Fayrfax, English Renaissance composer (died 1521)
1484: 1484–Julius Caesar Scaliger, Italian physician and scholar (died 1558)
1500: 1500–Alexander Ales, Scottish theologian and academic (died 1565)
1500: 1500   –Johann Stumpf, Swiss writer (died 1576)
1512: 1512–Henry FitzAlan, 19th Earl of Arundel, Chancellor of the University of Oxford (died 1580)
1516: 1516–Georg Fabricius, German poet, historian, and archaeologist (died 1571)
1564: 1564–William Shakespeare, English playwright and poet (died 1616)[22]
1598: 1598–Maarten Tromp, Dutch admiral (died 1653)
1621: 1621–William Penn, English admiral and politician (died 1670)
1628: 1628–Johannes Hudde, Dutch mathematician and politician (died 1704)
1661: 1661–Issachar Berend Lehmann, German-Jewish b

In [46]:
import requests
from bs4 import BeautifulSoup


def wikipedia(day, month):
    print(day, month)
    
    date = dt.datetime(2026, month, day)
    dated = date.strftime("%d %B %Y")
    print(dated)
    parts = dated.split()
    month = parts[1]
    print(f"MONTH inputed = {month}\nDAY inputed = {day}\n ")
    
    url = f"https://en.wikipedia.org/wiki/{month}_{day}"

    response = requests.get(url, headers={"User-Agent": "Mozilla/5.0"})
    soup = BeautifulSoup(response.text, "html.parser")

    births_header = soup.find("h2", id="Births")

    entries = []
    current_section = "Unknown"

    for sibling in births_header.parent.find_next_siblings():
        # Stop at next h2 section
        if sibling.get("class") and "mw-heading2" in sibling.get("class", []):
            break

        # Detect subsection headers (Pre-1600, 1601-1900, 1901-present)
        if sibling.get("class") and "mw-heading3" in sibling.get("class", []):
            current_section = sibling.get_text(strip=True).replace("[edit]", "").strip()
            continue

        for li in sibling.find_all("li"):
            text = li.get_text()
            parts = text.split("–", 1)
            if len(parts) == 2:
                year_str = parts[0].strip().split()[0].strip()
                
                year = int(year_str)
                entries.append({"section": current_section,"year": year,"text": text})
            
    fileName = f"{month}_{day}.txt"
    with open(fileName, 'w', encoding='utf-8') as file:
        current_sec = None
        for e in entries:
            if e["section"] != current_sec:
                current_sec = e["section"]
            print(f"{e['text']}")
            outcome = e['text'].replace("died", "d.")
            file.write(f"{outcome}\n")
    print(f"File saved as {month}_{day}.txt")
                
            
            
       
        # with open(fileName, 'w', ) as file:
        #     current_sec = None
        #     for e in entries:
        #         if e["section"] != current_sec:
        #             current_sec = e["section"]
        #             file.write(f"\n=== {current_sec} ===\n")
        #             print(f"\n=== {current_sec} ===")
        #         line = f"{e['year']}: {e['text'].strip()}\n"
        #         file.write(line)
            
day = 23
month = 4
print(wikipedia(day,month))

23 4
23 April 2026
MONTH inputed = April
DAY inputed = 23
 
1141 (probable)[19] – Malcolm IV of Scotland (died 1165)
1185 – Afonso II of Portugal (died 1223)[20]
1408 – John de Vere, 12th Earl of Oxford (died 1462)[21]
1420 – George of Poděbrady, King of Bohemia (died 1471)
1464 – Joan of France, Duchess of Berry (died 1505)
1464   – Robert Fayrfax, English Renaissance composer (died 1521)
1484 – Julius Caesar Scaliger, Italian physician and scholar (died 1558)
1500 – Alexander Ales, Scottish theologian and academic (died 1565)
1500   – Johann Stumpf, Swiss writer (died 1576)
1512 – Henry FitzAlan, 19th Earl of Arundel, Chancellor of the University of Oxford (died 1580)
1516 – Georg Fabricius, German poet, historian, and archaeologist (died 1571)
1564 – William Shakespeare, English playwright and poet (died 1616)[22]
1598 – Maarten Tromp, Dutch admiral (died 1653)
1621 – William Penn, English admiral and politician (died 1670)
1628 – Johannes Hudde, Dutch mathematician and politician (

In [33]:
import requests
from bs4 import BeautifulSoup
import datetime as dt
import time


def age(day, month, year):
    day = int(day)
    month = int(month)
    year = int(year)
    print(day, month, year)
    
    
    today = dt.datetime.now()
    today_month = today.month
    today_day = today.day
    today_year = today.year
    print(today, today_month, today_day, today_year)
    
  
    
    if today_month < month:
        age = today_year - year - 1
        print(f"Today {age} years old!")
    elif today_month == month and today_day < day: 
        age = today_year - year - 1
        print(f"Today {age} years old!")
    else:
        age = today_year - year
        print(f"Today {age} years old!") 
    
    
    
day = input("Day")
month =input("Month")
year = input("Year")
print(age(day,month, year))

27 5 2005
2026-04-26 22:32:29.892391 4 26 2026
Today 20 years old!
None


In [ ]:
import datetime as dt
date = "2 May 2013"
parts = date.split()


#strptime — parses a string into a datetime object
#strftime — formats a datetime object into a string
#newDate = dt.datetime.strptime(date)
# print(newDate)

In [45]:
from dateutil import parser
import datetime as dt
date = "2 May 2013"
newDate = parser.parse(date)
print(newDate)
# 2013-05-02 00:00:00
day = newDate.day
month = newDate.month
year = newDate.year
print(type(day))

2013-05-02 00:00:00
<class 'int'>


<!-- 23-04-1980
single

Today 45 years old!
Zodiac sign: Taurus
Percentage of people born on same day: 0.27 %
Historical events:
-0215 - A temple is built on the Capitoline Hill dedicated to Venus Erycina to commemorate the Roman defeat at Lake Trasimene.
0599 - Maya king Uneh Chan of Calakmul attacks rival city-state Palenque in southern Mexico, defeating queen Yohl Ik'nal and sacking the city.
0711 - Dagobert III succeeds his father King Childebert III as King of the Franks.
1014 - Battle of Clontarf: High King of Ireland Brian Boru defeats Viking invaders, but is killed in battle.
1016 - Edmund Ironside succeeds his father Æthelred the Unready as King of England.
1343 - St. George's Night Uprising commences in the Duchy of Estonia.
1348 - The founding of the Order of the Garter by King Edward III is announced on St. George's Day.
1500 - Portuguese explorer Pedro Alvarez Cabral reaches new coastline (Brazil).
1516 - The Munich Reinheitsgebot (regarding the ingredients of beer) takes effect in all of Bavaria.
1521 - Battle of Villalar: King Charles I of Spain defeats the Comuneros.
File saved as April_23.txt
Number one hit: Call Me - Blondie -->

In [ ]:
#pip install python-dateutil

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.0 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


23-04-1980
single

Today 45 years old!
Zodiac sign: Taurus
Percentage of people born on same day: 0.27 %
Historical events:
-0215 - A temple is built on the Capitoline Hill dedicated to Venus Erycina to commemorate the Roman defeat at Lake Trasimene.
0599 - Maya king Uneh Chan of Calakmul attacks rival city-state Palenque in southern Mexico, defeating queen Yohl Ik'nal and sacking the city.
0711 - Dagobert III succeeds his father King Childebert III as King of the Franks.
1014 - Battle of Clontarf: High King of Ireland Brian Boru defeats Viking invaders, but is killed in battle.
1016 - Edmund Ironside succeeds his father Æthelred the Unready as King of England.
1343 - St. George's Night Uprising commences in the Duchy of Estonia.
1348 - The founding of the Order of the Garter by King Edward III is announced on St. George's Day.
1500 - Portuguese explorer Pedro Alvarez Cabral reaches new coastline (Brazil).
1516 - The Munich Reinheitsgebot (regarding the ingredients of beer) takes effect in all of Bavaria.
1521 - Battle of Villalar: King Charles I of Spain defeats the Comuneros.
File saved as April_23.txt
Number one hit: Call Me - Blondie

In [ ]:
elif thread_type == "multi":
      
        thread1 = threading.Thread(target=age(day, month, year))

        thread2 = threading.Thread(target=zodiac(day, month))

        thread3 = threading.Thread(target=percentage(day, month))

        thread4 = threading.Thread(target=historical(day, month))

        thread5 = threading.Thread(target=wikipedia(day, month))

        thread6 = threading.Thread(target=one(day, month, year))

        thread1.start()
        thread2.start()
        thread3.start()
        thread4.start()
        thread5.start()
        thread6.start()

        thread1.join()
        thread2.join()
        thread3.join()
        thread4.join()
        thread5.join()
        thread6.join()

In [53]:
import json
import requests
from bs4 import BeautifulSoup
import time
import datetime as dt
import threading
import pandas as pd
from dateutil import parser


def zodiac(day, month):
    
    if month == 1 and day >= 20 and day <= 31 or month == 2 and day >= 1 and day <= 18:
        return f"Zodiac sign: Aquarius"
    elif  month == 2 and day >= 19 and day <= 29 or month == 3 and day >= 1 and day <= 20:
        return f"Zodiac sign: Pisces"
    elif  month == 3 and day >= 21 and day <= 31 or month == 4 and day >= 1 and day <= 19:
        return f"Zodiac sign: Aries"
    elif  month == 4 and day >= 20 and day <= 30 or month == 5 and day >= 1 and day <= 20:
        return f"Zodiac sign: Taurus"
    elif  month == 5 and day >= 21 and day <= 31 or month == 6 and day >= 1 and day <= 20:
        return f"Zodiac sign: Gemini"
    elif  month == 6 and day >= 21 and day <= 30 or month == 7 and day >= 1 and day <= 22:
        return f"Zodiac sign: Cancer"
    elif  month == 7 and day >= 23 and day <= 31 or month == 8 and day >= 1 and day <= 22:
        return f"Zodiac sign: Leo"
    elif  month == 8 and day >= 23 and day <= 31 or month == 9 and day >= 1 and day <= 22:
        return f"Zodiac sign: Virgo"
    elif  month == 9 and day >= 23 and day <= 30 or month == 10 and day >= 1 and day <= 22:
        return f"Zodiac sign: Libra"
    elif  month == 10 and day >= 23 and day <= 31 or month == 11 and day >= 1 and day <= 21:
        return f"Zodiac sign: Scorpio"
    elif  month == 11 and day >= 22 and day <= 30 or month == 12 and day >= 1 and day <= 21:
        return f"Zodiac sign: Sagittarius"
    elif  month == 12 and day >= 22 and day <= 31 or month == 1 and day >= 1 and day <= 19:
        return f"Zodiac sign: Capricorn"


def percentage(day, month):
    
    table = pd.read_csv("data.csv", sep = ",")

    maanden = table[table["date_of_month"] == day]
    lijst = maanden[maanden["month"] == month]

    sum_date_births = lijst["births"].sum()
    sum_births = table["births"].sum()
 
    percentage = (sum_date_births/ sum_births) * 100
    rounded = round(percentage, 2)

    return f"Percentage of people born on same day: {rounded} %"


def historical(day, month):
 
    url = f"https://api.api-ninjas.com/v1/historicalevents?month={month}&day={day}"
    headers = {"X-Api-Key" : "bncXEh6rAIsf7zRiKHR3BFJiilzJskTfH5u4Azus"}
    try:
        response = requests.get(url , headers=headers)

        if response.status_code == 200:
            data = response.json()
            
            output = "Historical events:"
            for x in data:
                year = x["year"]
                event = x["event"]
                outcome = f"\n{year} - {event}"
                output += outcome
            return output
    except requests.exceptions.RequestException as e:
        return f"Historical unavailable: {e}"

def one(day, month, year):

    day = int(day)
    month = int(month)
    year = int(year)
    
    date = dt.datetime(year, month, day).strftime("%d %B %Y")
    dated = date.split()
    month = dated[1].lower()
    
    url = f"https://www.onthisday.com/date/{year}/{month}/{day}"
    
    headers = {"User-Agent": "Anthony"}
    response = requests.get(url, headers=headers )
 
    soup = BeautifulSoup(response.text, "html.parser")
    evenementen = soup.find_all("li", class_="event")

    for e in evenementen:
        text = e.get_text(" ")
        image = e.find("img")
        
        if image:
            image.get("src")   
            if image.get("alt") == "US":
                output = text[11:].replace("  "," ")
                return f"Number one hit:{output}"
                break


def wikipedia(day, month):
    
    date = dt.datetime(2026, month, day)
    dated = date.strftime("%d %B %Y")
    parts = dated.split()
    month = parts[1]

    url = f"https://en.wikipedia.org/wiki/{month}_{day}"

    response = requests.get(url, headers={"User-Agent": "Mozilla/5.0"})
    soup = BeautifulSoup(response.text, "html.parser")

    births_header = soup.find("h2", id="Births")

    entries = []
    current_section = "Unknown"
    if births_header is None:
       return "Wikipedia data unavailable"
   
    for sibling in births_header.parent.find_next_siblings():
        if sibling.get("class") and "mw-heading2" in sibling.get("class", []):
            break

        if sibling.get("class") and "mw-heading3" in sibling.get("class", []):
            current_section = sibling.get_text(strip=True).replace("[edit]", "").strip()
            continue

        for li in sibling.find_all("li"):
            text = li.get_text()
            parts = text.split("–", 1)
            if len(parts) == 2:
                year_str = parts[0].strip().split()[0].strip()
                
                year = int(year_str)
                entries.append({"section": current_section,"year": year,"text": text})
            
    fileName = f"{month}_{day}.txt"
    with open(fileName, 'w', encoding='utf-8') as file:
        current_sec = None
        for e in entries:
            if e["section"] != current_sec:
                current_sec = e["section"]

            outcome = e['text'].replace("died", "d.")
            file.write(f"{outcome}\n")
    return f"File saved as {fileName}"


def age(day, month, year):
    day = int(day)
    month = int(month)
    year = int(year)

    today = dt.datetime.now()
    today_month = today.month
    today_day = today.day
    today_year = today.year

    if today_month < month:
        age = today_year - year - 1
        return f"Today {age} years old!"
    elif today_month == month and today_day < day: 
        age = today_year - year - 1
        return f"Today {age} years old!"
    else:
        age = today_year - year
        return f"Today {age} years old!"



def main():
    date = input("Enter Date:")
    newDate = parser.parse(date)
    
    day = newDate.day
    month = newDate.month
    year = newDate.year


    
    
    thread_type = input("Enter thread(single / multi)")
    if thread_type == "single":
        print(age(day, month, year))
        print(zodiac(day, month))
        print(percentage(day, month))
        print(historical(day, month))
        print(wikipedia(day, month))
        print(one(day, month, year))
    elif thread_type == "multi":
      
        thread1 = threading.Thread(target=age, args=(day, month, year))
        thread2 = threading.Thread(target=zodiac, args=(day, month))
        thread3 = threading.Thread(target=percentage, args=(day, month))
        thread4 = threading.Thread(target=historical, args=(day, month))
        thread5 = threading.Thread(target=wikipedia, args=(day, month))
        thread6 = threading.Thread(target=one, args=(day, month, year))

        thread1.start()
        thread2.start()
        thread3.start()
        thread4.start()
        thread5.start()
        thread6.start()

        thread1.join()
        thread2.join()
        thread3.join()
        thread4.join()
        thread5.join()
        thread6.join()



if __name__ == "__main__": 
     print(main())



None


In [ ]:
import json
import requests
from bs4 import BeautifulSoup
import time
import datetime as dt
import threading
import pandas as pd
from dateutil import parser


def zodiac(day, month):
    if month == 1 and day >= 20 and day <= 31 or month == 2 and day >= 1 and day <= 18:
        return f"Zodiac sign: Aquarius"
    elif month == 2 and day >= 19 and day <= 29 or month == 3 and day >= 1 and day <= 20:
        return f"Zodiac sign: Pisces"
    elif month == 3 and day >= 21 and day <= 31 or month == 4 and day >= 1 and day <= 19:
        return f"Zodiac sign: Aries"
    elif month == 4 and day >= 20 and day <= 30 or month == 5 and day >= 1 and day <= 20:
        return f"Zodiac sign: Taurus"
    elif month == 5 and day >= 21 and day <= 31 or month == 6 and day >= 1 and day <= 20:
        return f"Zodiac sign: Gemini"
    elif month == 6 and day >= 21 and day <= 30 or month == 7 and day >= 1 and day <= 22:
        return f"Zodiac sign: Cancer"
    elif month == 7 and day >= 23 and day <= 31 or month == 8 and day >= 1 and day <= 22:
        return f"Zodiac sign: Leo"
    elif month == 8 and day >= 23 and day <= 31 or month == 9 and day >= 1 and day <= 22:
        return f"Zodiac sign: Virgo"
    elif month == 9 and day >= 23 and day <= 30 or month == 10 and day >= 1 and day <= 22:
        return f"Zodiac sign: Libra"
    elif month == 10 and day >= 23 and day <= 31 or month == 11 and day >= 1 and day <= 21:
        return f"Zodiac sign: Scorpio"
    elif month == 11 and day >= 22 and day <= 30 or month == 12 and day >= 1 and day <= 21:
        return f"Zodiac sign: Sagittarius"
    elif month == 12 and day >= 22 and day <= 31 or month == 1 and day >= 1 and day <= 19:
        return f"Zodiac sign: Capricorn"


def percentage(day, month):
    table = pd.read_csv("data.csv", sep=",")
    maanden = table[table["date_of_month"] == day]
    lijst = maanden[maanden["month"] == month]
    sum_date_births = lijst["births"].sum()
    sum_births = table["births"].sum()
    percentage = (sum_date_births / sum_births) * 100
    rounded = round(percentage, 2)
    return f"Percentage of people born on same day: {rounded} %"


def historical(day, month):
    url = f"https://api.api-ninjas.com/v1/historicalevents?month={month}&day={day}"
    headers = {"X-Api-Key": "bncXEh6rAIsf7zRiKHR3BFJiilzJskTfH5u4Azus"}
    try:
        response = requests.get(url, headers=headers)
        if response.status_code == 200:
            data = response.json()
            output = "Historical events:"
            for x in data:
                year = x["year"]
                event = x["event"]
                output += f"\n{year} - {event}"
            return output
    except requests.exceptions.RequestException:
        return "Historical events unavailable (network blocked)"


def one(day, month, year):
    day = int(day)
    month = int(month)
    year = int(year)

    date = dt.datetime(year, month, day).strftime("%d %B %Y")
    dated = date.split()
    month = dated[1].lower()

    url = f"https://www.onthisday.com/date/{year}/{month}/{day}"
    headers = {"User-Agent": "Anthony"}
    response = requests.get(url, headers=headers)
    soup = BeautifulSoup(response.text, "html.parser")
    evenementen = soup.find_all("li", class_="event")

    for e in evenementen:
        text = e.get_text(" ")
        image = e.find("img")
        if image:
            image.get("src")
            if image.get("alt") == "US":
                output = text[11:].replace("  ", " ")
                return f"Number one hit:{output}"


def wikipedia(day, month):
    date = dt.datetime(2026, month, day)
    dated = date.strftime("%d %B %Y")
    parts = dated.split()
    month = parts[1]

    url = f"https://en.wikipedia.org/wiki/{month}_{day}"
    response = requests.get(url, headers={"User-Agent": "Mozilla/5.0"})
    soup = BeautifulSoup(response.text, "html.parser")

    births_header = soup.find("h2", id="Births")
    if births_header is None:
        return "Wikipedia data unavailable"

    entries = []
    current_section = "Unknown"

    for sibling in births_header.parent.find_next_siblings():
        if sibling.get("class") and "mw-heading2" in sibling.get("class", []):
            break
        if sibling.get("class") and "mw-heading3" in sibling.get("class", []):
            current_section = sibling.get_text(strip=True).replace("[edit]", "").strip()
            continue
        for li in sibling.find_all("li"):
            text = li.get_text()
            parts = text.split("–", 1)
            if len(parts) == 2:
                year_str = parts[0].strip().split()[0].strip()
                try:
                    year = int(year_str)
                    entries.append({"section": current_section, "year": year, "text": text})
                except ValueError:
                    pass

    fileName = f"{month}_{day}.txt"
    with open(fileName, 'w', encoding='utf-8') as file:
        current_sec = None
        for e in entries:
            if e["section"] != current_sec:
                current_sec = e["section"]
            outcome = e['text'].replace("died", "d.")
            file.write(f"{outcome}\n")
    return f"File saved as {fileName}"


def age(day, month, year):
    day = int(day)
    month = int(month)
    year = int(year)

    today = dt.datetime.now()
    today_month = today.month
    today_day = today.day
    today_year = today.year

    if today_month < month:
        age = today_year - year - 1
        return f"Today {age} years old!"
    elif today_month == month and today_day < day:
        age = today_year - year - 1
        return f"Today {age} years old!"
    else:
        age = today_year - year
        return f"Today {age} years old!"


def main():
    date = input()
    newDate = parser.parse(date)

    day = newDate.day
    month = newDate.month
    year = newDate.year

    thread_type = input()

    if thread_type == "single":
        print(age(day, month, year))
        print(zodiac(day, month))
        print(percentage(day, month))
        print(historical(day, month))
        print(wikipedia(day, month))
        print(one(day, month, year))

    elif thread_type == "multi":
        results = {}

        def run(func, key, *args):
            results[key] = func(*args)

        threads = [
            threading.Thread(target=run, args=(age,        "age",  day, month, year)),
            threading.Thread(target=run, args=(zodiac,     "zodiac", day, month)),
            threading.Thread(target=run, args=(percentage, "pct",  day, month)),
            threading.Thread(target=run, args=(historical, "hist", day, month)),
            threading.Thread(target=run, args=(wikipedia,  "wiki", day, month)),
            threading.Thread(target=run, args=(one,        "one",  day, month, year)),
        ]

        for t in threads:
            t.start()
        for t in threads:
            t.join()

        for val in results.values():
            print(val)


if __name__ == "__main__":
    main()

Today 22 years old!
Zodiac sign: Taurus
Percentage of people born on same day: 0.27 %
Number one hit: Yeah! - Usher Featuring Lil Jon & Ludacris
File saved as April_23.txt
Historical events:
-0215 - A temple is built on the Capitoline Hill dedicated to Venus Erycina to commemorate the Roman defeat at Lake Trasimene.
0599 - Maya king Uneh Chan of Calakmul attacks rival city-state Palenque in southern Mexico, defeating queen Yohl Ik'nal and sacking the city.
0711 - Dagobert III succeeds his father King Childebert III as King of the Franks.
1014 - Battle of Clontarf: High King of Ireland Brian Boru defeats Viking invaders, but is killed in battle.
1016 - Edmund Ironside succeeds his father Æthelred the Unready as King of England.
1343 - St. George's Night Uprising commences in the Duchy of Estonia.
1348 - The founding of the Order of the Garter by King Edward III is announced on St. George's Day.
1500 - Portuguese explorer Pedro Alvarez Cabral reaches new coastline (Brazil).
1516 - The Mu